WebViewDataTest fails on API 26 Pixel emulator

Stack trace:
```
java.lang.AssertionError: Expected file 'perfd' to be in the app's data dir whitelist
at org.junit.Assert.fail(Assert.java:88)
at org.junit.Assert.assertTrue(Assert.java:41)
at org.mozilla.focus.activity.WebViewDataTest.DeleteWebViewDataTest(WebViewDataTest.java:205)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
```

It seems there are a few files in the data/ directory that don't exist on my API 23 emulator, which is why I didn't catch this sooner.

We should also verify that these files, if we whitelist them, aren't leaking browsing state. Otherwise, we should delete them!

--- a/app/src/androidTest/java/org/mozilla/focus/activity/WebViewDataTest.java
+++ b/app/src/androidTest/java/org/mozilla/focus/activity/WebViewDataTest.java
@@ -202,7 +202,8 @@ public class WebViewDataTest {
                 continue;
             }
 
-            assertTrue("Check '" + name + "' is in whitelist", WHITELIST_DATA_DIR_CONTENTS.contains(name));
+            assertTrue("Expected file '" + name + "' to be in the app's data dir whitelist",
+                    WHITELIST_DATA_DIR_CONTENTS.contains(name));
         }
 
         assertNoTraces(dataDir);

--- a/app/src/androidTest/java/org/mozilla/focus/activity/WebViewDataTest.java
+++ b/app/src/androidTest/java/org/mozilla/focus/activity/WebViewDataTest.java
@@ -54,7 +54,19 @@ public class WebViewDataTest {
             "app_dxmaker_cache",
             "telemetry",
             "databases",
-            "app_webview"
+            "app_webview",
+
+            // Android Studio will inject these files if you:
+            // - Build with Android Studio 3.0+
+            // - Have opened the "Android Profiler" tab at least once since the AS process started
+            // - Run on an API 26+ device (or maybe when explicitly enabling advanced debugging on older devices)
+            //
+            // This should only affect local builds and we don't want to risk breaking the profiler so
+            // we whitelist them. Additional details around when these files are added can be found in:
+            //   https://github.com/mozilla-mobile/focus-android/issues/1842#issuecomment-348038392
+            "libperfa_x86.so",
+            "perfa.jar",
+            "perfd"
     );
 
     // We expect those folders to exist but they should be empty.

--- a/app/src/androidTest/java/org/mozilla/focus/activity/WebViewDataTest.java
+++ b/app/src/androidTest/java/org/mozilla/focus/activity/WebViewDataTest.java
@@ -47,7 +47,7 @@ import static org.mozilla.focus.fragment.FirstrunFragment.FIRSTRUN_PREF;
  */
 @RunWith(AndroidJUnit4.class)
 public class WebViewDataTest {
-    private static final List<String> WHITELIST = Arrays.asList(
+    private static final List<String> WHITELIST_DATA_DIR_CONTENTS = Arrays.asList(
             "cache", // We assert that this folder is empty
             "code_cache",
             "shared_prefs",
@@ -202,7 +202,7 @@ public class WebViewDataTest {
                 continue;
             }
 
-            assertTrue("Check '" + name + "' is in whitelist", WHITELIST.contains(name));
+            assertTrue("Check '" + name + "' is in whitelist", WHITELIST_DATA_DIR_CONTENTS.contains(name));
         }
 
         assertNoTraces(dataDir);

